### DISCUSSION TITLE
Alpine should not work outside of x-data element

### DISCUSSION THREAD

**Discussion by hissy** [alpinejs/alpine#4108] [7. Bugs (archived)]
It seems `x-bind` virtual attributes are copied to actual attributes.

For example, if you have an element that has an attribute its key started with `:`,

```html
<div :custom-attribute="{}">Test Element</div>
```

its value will be copied to another attribute like this:

```html
<div :custom-attribute="{}" custom-attribute="[object Object]">Test Element</div>
```

You can see the copied attribute from the developer tool.

![Screenshot 2024-03-23 at 9 58 46](https://github.com/alpinejs/alpine/assets/514294/e2b895c9-22e5-4466-b6c5-6d3cf2bc24bd)

https://jsfiddle.net/hissy/5px61c4q/1/

Is this an expected result? If so, how to stop it?
I'd like to prevent it because it occurs error when using alpine.js with vue.js on the same page.

**Comment by hissy** [comment]
Also, my jsfiddle doesn't contains any x-init or x-data attributes. I think this may be a bug.

> By default, Alpine will crawl and initialize the entire DOM tree of an element containing x-init or x-data.
https://alpinejs.dev/directives/ignore

**Comment by ekwoka** [comment]
What would you want it to do with the custom attribute? Where should it put it? If it's an attribute...it needs to be an attribute on the element...otherwise it's not an attribute...

And why do you have Vue and Alpine working on the same elements?

**Comment by hissy** [comment]
@ekwoka 

> What would you want it to do with the custom attribute?

You can pass some values with custom attributes to Vue components.

> And why do you have Vue and Alpine working on the same elements?

I don't want to apply Alpine and Vue on the same elements.
Alpine affects Vue components even if those are not in the tree of elements applied x-init or x-data.

**Comment by ekwoka** [comment]
I think this is caused by how Vue replaces the elements on the page and Alpine initializes all added elements as Alpine even if they aren't inside a root.

That would be a bug.

:thinking:


But it will still add the real attribute. That's just how it works.

For your specific case, you should use the long form. V-bind: and x-bind exclusively, same with v-on and x-on. Not the shorthands

**Comment by hissy** [comment]
@ekwoka 

Thank you for your advice. I was temporarily fixed by adding the x-ignore attribute to Vue components. I'll try to using long form also.

> But it will still add the real attribute. That's just how it works.

Now I understand. However, I think Alpine should not add attributes outside of x-data element. Yeah, that may be a bug.

**Reply by ekwoka** [reply]
Yes, it shouldn't and doesn't under most cases.

I think elements that swap in afterwards can.

There may also be an x-data you are forgetting about. Normally Vue won't leave the : directives on elements...

**Comment by hissy** [comment]
I beleave this is a bug, so I've changed the title and category.
